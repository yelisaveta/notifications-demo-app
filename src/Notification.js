import React, { Component } from "react";
import PropTypes from "prop-types";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Button from "@material-ui/core/Button";
import DeleteIcon from "@material-ui/icons/Delete";
import { withStyles } from "@material-ui/core/styles";

import LikeButton from "./LikeButton";
import NotificationHeader from "./NotificationHeader";

const styles = theme => ({
  card: {
    width: "100%"
  },
  rightIcon: {
    marginLeft: theme.spacing.unit
  }
});

export class Notification extends Component {
  handleDelete = id => {
    return () => {
      this.props.onDelete(id);
    };
  };

  handleLike = id => {
    return () => {
      this.props.onLike(id);
    };
  };

  render() {
    const { id, title, message, liked, classes } = this.props;
    return (
      <Card className={classes.card} data-testid={`notification-id-${id}`}>
        <NotificationHeader title={title} liked={liked} />
        <CardContent>{message}</CardContent>
        <CardActions>
          <LikeButton onLike={this.handleLike(id)} />
          <Button
            variant="contained"
            color="primary"
            size="medium"
            onClick={this.handleDelete(id)}
          >
            Delete
            <DeleteIcon className={classes.rightIcon} />
          </Button>
        </CardActions>
      </Card>
    );
  }
}

Notification.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  liked: PropTypes.bool,
  onDelete: PropTypes.func.isRequired,
  onLike: PropTypes.func.isRequired,
  classes: PropTypes.shape({
    card: PropTypes.string.isRequired,
    rightIcon: PropTypes.string.isRequired
  })
};

export default withStyles(styles)(Notification);
