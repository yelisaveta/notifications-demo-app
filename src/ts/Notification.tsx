import * as React from "react";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Button from "@material-ui/core/Button";
import DeleteIcon from "@material-ui/icons/Delete";
import { withStyles } from "@material-ui/core/styles";
import { Theme } from "@material-ui/core/styles/createMuiTheme";

import LikeButton from "../LikeButton";
import NotificationHeader from "../NotificationHeader";

export interface NotificationProps {
  id: string;
  title: string;
  message: string;
  liked: boolean;
  onDelete: Function;
  onLike: Function;
  classes: {
    card: string;
    rightIcon: string;
  };
}

const styles = (theme: Theme) => ({
  card: {
    width: "100%"
  },
  rightIcon: {
    marginLeft: theme.spacing.unit,
    marginRight: 20
  }
});

export class Notification extends React.Component<NotificationProps, {}> {
  handleDelete = (id: string) => {
    return () => {
      this.props.onDelete(id);
    };
  };

  handleLike = (id: string) => {
    return () => {
      this.props.onLike(id);
    };
  };

  render() {
    const { id, title, message, liked, classes } = this.props;
    return (
      <Card className={classes.card} data-testid={`notification-id-${id}`}>
        <NotificationHeader title={title} liked={liked} />
        <CardContent>{message}</CardContent>
        <CardActions>
          <LikeButton onLike={this.handleLike(id)} />
          <Button
            variant="contained"
            color="primary"
            size="small"
            onClick={this.handleDelete(id)}
          >
            Delete
            <DeleteIcon className={classes.rightIcon} />
          </Button>
        </CardActions>
      </Card>
    );
  }
}

export default withStyles(styles)(Notification);
