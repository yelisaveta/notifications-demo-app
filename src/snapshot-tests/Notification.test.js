import React from "react";
import renderer from "react-test-renderer";
import Notification from "../Notification";

const props = {
  id: "notification-test-id",
  title: "my test title",
  message: "my test message",
  onDelete: jest.fn(),
  onLike: jest.fn()
};

// Generated snapshot test is too big and contains auto-generated Material UI class names
describe("Notification full snapshot test", () => {
  it("renders a Notification", () => {
    const tree = renderer.create(<Notification {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
