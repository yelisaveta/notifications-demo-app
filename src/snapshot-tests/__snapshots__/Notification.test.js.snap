// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification full snapshot test renders a Notification 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiPaper-rounded-5 MuiCard-root-3 Notification-card-1"
  data-testid="notification-id-notification-test-id"
>
  <div
    className="MuiCardHeader-root-31"
  >
    <div
      className="MuiCardHeader-content-34"
    >
      <span
        className="MuiTypography-root-37 MuiTypography-headline-42 MuiCardHeader-title-35"
      >
        my test title
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root-73"
  >
    my test message
  </div>
  <div
    className="MuiCardActions-root-74"
  >
    <button
      className="MuiButtonBase-root-103 MuiButton-root-77 MuiButton-outlined-85"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-78"
      >
        Like
      </span>
      <span
        className="MuiTouchRipple-root-115"
      />
    </button>
    <button
      className="MuiButtonBase-root-103 MuiButton-root-77 MuiButton-contained-88 MuiButton-containedPrimary-89 MuiButton-raised-91 MuiButton-raisedPrimary-92 MuiCardActions-action-76"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-78"
      >
        Delete
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-106 Notification-rightIcon-2"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-115"
      />
    </button>
  </div>
</div>
`;
